### DISCUSSION TITLE
Flask Assets Source Code

### DISCUSSION THREAD

**Discussion by HadarSha** [vanna-ai/vanna#515] [Q&A]
Hello everyone,

I am thrilled by this project and the great work by @zainhoda - cheers!

One thing that has been a limitation for me has been the lack of Flask Assets Source Code, as they seem to be only available in some minified (or otherwise obfuscated) form, either embedded in the Python code like here:
https://github.com/vanna-ai/vanna/blob/main/src/vanna/flask/assets.py

Or the seemingly outdated, non-maintained Flask app here, which does not seem maintainable either:
https://github.com/vanna-ai/vanna-flask/tree/main/static/assets

In other words - I would love to have access to the source CSS and JavaScript in a manner that would allow me to not only maintain, but also build and extend upon them. 

I really like the Flask Front-end that @zainhoda has put together, but if I want to add, say, a toggle or a switch, I have a very hard time doing that with the HTML being very concise and almost all of the logic packed in the minified .css and .js files

I also saw [a similar discussion](https://github.com/vanna-ai/vanna/discussions/459) that was left unanswered. 

What do you think? Could we get the ball rolling somehow? I imagine there could be loads of improvements made to the Flask UI, which is quite impressive already! Cheers

**Comment by zawalich** [comment]
Big up for this - better than replacing content of `html_content`, `css_content`, `js_content` objects with Pythonic `.replace()`.
